be/src/olap/schema_change.cpp

```diff
@@ -1332,9 +1332,12 @@ OLAPStatus SchemaChangeHandler::_do_process_alter_tablet_v2(const TAlterTabletRe
         if (res != OLAP_SUCCESS) {
             break;
         }
-        res = _validate_alter_result(new_tablet, request);
     } while(0);
 
+    // _validate_alter_result should be outside the above while loop.
+    // to avoid requiring the header lock twice.
+    res = _validate_alter_result(new_tablet, request);
```

Comment:
If the previous step failed, this step will override the result

Reply:
Add a res check before this